tests/test_other.py

```diff
@@ -12155,3 +12155,27 @@ def test_print_map(self):
     self.assertContained('hello_world.o:(__original_main)', out)
     out2 = self.run_process([EMCC, 'hello_world.o', '-Wl,-M'], stdout=PIPE).stdout
     self.assertEqual(out, out2)
+
+  def test_rust_gxx_personality_v0(self):
+    create_file('main.cpp', r'''
+      #include <stdio.h>
+      #include <stdint.h>
+      extern "C" {
+        int __gxx_personality_v0(int version, void* actions, uint64_t exception_class, void* exception_object, void* context);
+        int main() {
+          __gxx_personality_v0(0, NULL, 0, NULL, NULL);
+          return 0;
+        }
+      }
+    ''')
+    self.run_process([EMXX, '-c', 'main.cpp'])
+
+    self.run_process([EMXX, '-o', 'main.js', 'main.o', '-lc++abi'] + self.get_emcc_args())
+
+    try:
+      self.do_run('main.js', no_build=True)
+      raise RuntimeError('should not have passed')
+    except AssertionError as e:
+      err = e
+    assert "Aborted(native code called abort())" in err.args[0]
+    del err
```

Comment:
You should be able to do this with:

`self.do_run('main.cpp', 'Aborted(native code called abort())', assert_returncode=NON_ZERO)`

I don't you need a separate compile step, of the explict `-lc++abi` ... or the try/catch.

Reply:
Why does it need that?

Reply:
`self.emcc_args.append('-fwasm-exceptions')`?

Reply:
Test fails without it.

Reply:
Or do you mean to move the definition of the stub outside of the `#ifdef`?

Reply:
If we want to ensure this works both with and without Emscripten EH, shouldn't we test this both with `-fexceptions` and without?